### PR TITLE
Added projector Python wrapper

### PR DESCRIPTION
**Pull request by ahcorde**
# 🎉 New feature

## Summary

Added projector python wrapper

Related to this PR https://github.com/gazebosim/sdformat/pull/1277

## Test it

```bash
colcon test --merge-install --event-handlers console_direct+ --packages-select sdformat13 --ctest-args -R pyProjector
```

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: python/src/sdf/pyLink.cc [hunk 1]
```diff
@@ -177,6 +195,9 @@ void defineLink(pybind11::object module)
     // .def("clear_particle_emitters",
     //      &sdf::Link::ClearParticleEmitters,
     //      "Remove all particle emitters")
+    .def("clear_projector",
+         &sdf::Link::ClearProjectors,
+         "Remove all projector")
```
Comment (iche033) [inline python/src/sdf/pyLink.cc:200]:
```suggestion
         "Remove all projectors")
```

File: python/src/sdf/pyLink.cc [hunk 2]
```diff
@@ -162,6 +177,9 @@ void defineLink(pybind11::object module)
     // .def("AddParticleEmitter",
     //      &sdf::Link::AddParticleEmitter,
     //      "Add a particle emitter to the link.")
+    .def("add_projector",
+         &sdf::Link::AddProjector,
+         "Add a particle projector to the link.")
```
Comment (iche033) [inline python/src/sdf/pyLink.cc:182]:
```suggestion
         "Add a projector to the link.")
```

File: python/src/sdf/pyLink.cc [hunk 3]
```diff
@@ -87,6 +88,20 @@ void defineLink(pybind11::object module)
            &sdf::Link::LightByName),
          pybind11::return_value_policy::reference_internal,
          "Get a mutable light based on a name.")
+    .def("projector_count", &sdf::Link::ProjectorCount,
+         "Get the number of projector.")
```
Comment (iche033) [inline python/src/sdf/pyLink.cc:92]:
```suggestion
         "Get the number of projectors.")
```

**Review by iche033 — COMMENTED** [review]
thanks for taking care of this! Just minor comments.

File: python/src/sdf/pyLink.cc (hunk 3 above)
Comment (scpeters) [reply python/src/sdf/pyLink.cc:92]: +1

File: python/src/sdf/pyLink.cc (hunk 2 above)
Comment (scpeters) [reply python/src/sdf/pyLink.cc:182]: +1

File: python/src/sdf/pyLink.cc (hunk 1 above)
Comment (scpeters) [reply python/src/sdf/pyLink.cc:200]: +1

**Review by iche033 — APPROVED** [review]

**Pull request merged**